src/ui/chart-list.jsx

```diff
@@ -0,0 +1,241 @@
+import TrackerState from '../services/tracker-state';
+
+import FoundAtTooltip from './found-at-tooltip';
+import Images from './images';
+import KeyDownWrapper from './key-down-wrapper';
+import Tooltip from './tooltip';
+
+class ChartList extends React.PureComponent {
+  static NUM_ROWS = 20;
+
+  mapChart(chart) {
+    const {
+      openedChartForIsland,
+      trackerState,
+      updateChartMapping,
+    } = this.props;
+
+    if (_.isNil(chart)) {
+      return null;
+    }
+
+    const itemCount = trackerState.getItemValue(chart);
+
+    const mappedIslandForChart = trackerState.getIslandForChart(chart);
+    const isChartMapped = !_.isNil(mappedIslandForChart);
+
+    const notInteractiveClassName = isChartMapped ? 'detail-not-interactive' : '';
+
+    let color;
+    if (isChartMapped) {
+      color = LogicCalculation.LOCATION_COLORS.CHECKED_LOCATION;
+    } else if (itemCount === 1) {
+      color = LogicCalculation.LOCATION_COLORS.AVAILABLE_LOCATION;
+    } else {
+      color = LogicCalculation.LOCATION_COLORS.UNAVAILABLE_LOCATION;
+    }
+
+    const updateChartMappingFunc = (event) => {
+      event.stopPropagation();
+      if (!isChartMapped) {
+        updateChartMapping(chart, openedChartForIsland);
+      }
+    };
+
+    const chartElement = (
+      <div
+        className={`detail-span ${notInteractiveClassName} ${color} font-smallest`}
+        onClick={updateChartMappingFunc}
+        onContextMenu={updateChartMappingFunc}
+        onKeyDown={KeyDownWrapper.onSpaceKey(updateChartMappingFunc)}
+        role="button"
+        tabIndex="0"
+      >
+        {chart}
+      </div>
+    );
+
+    let chartContent;
+    if (isChartMapped) {
+      const tooltip = (
+        <div className="tooltip">
+          <div className="tooltip-title">Chart Leads To</div>
+          <div>{mappedIslandForChart}</div>
+        </div>
+      );
+
+      chartContent = (
+        <Tooltip tooltipContent={tooltip}>
+          {chartElement}
+        </Tooltip>
+      );
+    } else {
+      chartContent = chartElement;
+    }
+
+    return <td key={chart}>{chartContent}</td>;
+  }
+
+  chart(chartName, showLocationTooltip = true) {
+    if (_.isNil(chartName)) {
+      return null;
+    }
+
+    const {
+      decrementItem,
+      incrementItem,
+      spheres,
+      trackerState,
+      trackSpheres,
+    } = this.props;
+
+    const itemCount = trackerState.getItemValue(chartName);
+
+    const color = itemCount === 1
+      ? LogicCalculation.LOCATION_COLORS.CHECKED_LOCATION
+      : LogicCalculation.LOCATION_COLORS.AVAILABLE_LOCATION;
+
+    let locations = [];
+    if (showLocationTooltip && trackSpheres) {
+      locations = trackerState.getLocationsForItem(chartName);
+    }
+
+    const incrementItemFunc = (event) => {
+      event.stopPropagation();
+
+      incrementItem(chartName);
+    };
+
+    const decrementItemFunc = (event) => {
+      event.preventDefault();
+
+      decrementItem(chartName);
+    };
```

Comment:
We don't need a `decrementItemFunc` here.

Also, we need to handle an additional case: If we're unchecking a chart in the chart list, it should also remove the chart to island mapping for that chart.